src/Core/tests/UnitTests/Views/BorderTests.cs

```diff
@@ -29,6 +30,23 @@ public void TestBorderPropagateBindingContext()
 			Assert.True(border.BindingContext == bindingContext);
 		}
 
+		[Fact]
+		public void TestStrokeShapeBindingContext()
+		{
+			var context = new object();
+
+			var parent = new Border
+			{
+				BindingContext = context
+			};
+
+			var strokeShape = new Rectangle();
+
+			parent.StrokeShape = strokeShape;
+
+			Assert.Same(context, ((Rectangle)parent.StrokeShape).BindingContext);
+		}
```

Comment:
This needs a test proving there is no memory leak now.

Reply:
This one had a good example: https://github.com/dotnet/maui/pull/13806

https://github.com/dotnet/maui/blob/9e5785a3c65f04ea7114b0c8c8e174c745045441/src/Controls/tests/Core.UnitTests/VisualElementTests.cs#L147-L160

We make sure that the `Clip` (the publisher) doesn't hold onto the `VisualElement` (the subscriber).

I would start adding one of these every time you see `+=`.